Chance boost kill

WithersChat opened the pull request:
## What
Complete removal of chance boosting from GTm

## Implementation Details
A lot of code deletion and some modifications

## Outcome
Chance Boosting is kill

## Additional Information
Screenshots will be added once I'm done implementing this

## Potential Compatibility Issues
This will definitely break some addons and packs.

WithersChat committed:
And so it begins [DOES NOT COMPILE]

- Commented out ChanceBoostFunction
- Edited ChanceLogic to stop relying on it
- Edited RecipeRunner to match the changes above
- DOES NOT COMPILE, CHANGE IN PROGRESS

WithersChat committed:
Why am I doing this to myself? [DOES NOT COMPILE]

- Removed chance logic from Content.java
- Edited GTOreByProductWidget with truncated method signature
- DOES NOT COMPILE, CHANGE IN PROGRESS

WithersChat committed:
Why is this lowkey addictive? [DOES NOT COMPILE]

- Cleaned up GTRecipeWidget and users
- Cleaned up RecipeCapability and inheritors
- Removed chanceFunction from GTRecipeType
- DOES NOT COMPILE, CHANGE IN PROGRESS

WithersChat committed:
Lang my archenemy! [DOES NOT COMPILE¯

- Cleaned up some lang so my double-shift search works better
- DOES NOT COMPILE, CHANGE IN PROGRESS

screret commented:
this could be done by just changing the default logic
that way people who want to keep it can still keep it as well

WithersChat committed:
Mission put on hold until 1.8 [DOES NOT COMPILE]

- Just pushed some more stuff I had done
- DOES NOT COMPILE, CHANGE IN PROGRESS

screret commented:
superseded by #3026
